packages/api/lib/rulesHelpers.js

```diff
@@ -6,6 +6,31 @@ const { removeNilProperties } = require('@cumulus/common/util');
 const { handleScheduleEvent } = require('../lambdas/sf-scheduler');
 const Rule = require('../models/rules');
 
+const filterRulesbyCollection = (rules, collection) => rules.filter(
+  (rule) => {
+    // Match as much collection info as we found in the message
+    const nameMatch = collection.name
+      ? get(rule, 'collection.name') === collection.name
+      : true;
+    const versionMatch = collection.version
+      ? get(rule, 'collection.version') === collection.version
+      : true;
+    return nameMatch && versionMatch;
+  }
+);
+
+const getMaxTimeoutForRules = (rules) => rules.reduce(
```

Comment:
Very nitpicky, but something like this seems a bit simpler to me:
```
const values = rules.map((rule) => get(rule, 'meta.visibilityTimeout')).filter((t) => t);
values.reduce((a, b) => Math.max(a, b));
```

Reply:
The refactor isn't quite equivalent. Because of the `filter` operation, rules without a timeout get filtered out, so if there was only one rule without a timeout, then there would be an empty array for `.reduce` to operate on. Also, I *think* that we want to return `undefined` if none of the rules have a visibility timeout, because a value of `undefined` won't be used by the consumer when it goes to fetch messages whereas any numeric value would be.

The tests for `getMaxTimeoutForRules` fail with this change:

https://github.com/nasa/cumulus/pull/1638/files?file-filters%5B%5D=.js#diff-023361c061b8df15ce4e31d181c7e2adR108